Sources/AsyncKit/ConnectionPool.swift

```diff
@@ -82,11 +90,17 @@ public final class ConnectionPool<Source> where Source: ConnectionPoolSource  {
     private var available: [Source.Connection]
     
     /// Current active connection count.
-    private var activeConnections: Int
+    private var activeConnections: Atomic<Int>
```

Comment:
Why did this become atomic? If we synchronize all access through it to `lock`, `Atomic` shouldn't be necessary.

Reading through the rest of the changes, I guess you try to avoid acquiring a lock when accessing `activeConnections`. I would suggest just acquiring the lock then and waiting until performance issues are reported. Performance will probably never be an issue even when synchronizing access to `activeConnections`, so trying to avoid that lock would be premature optimization that could lead to race conditions later on.

Reply:
Good point, i'll just lock instead. 👍